### PR TITLE
Added task solution

### PR DESCRIPTION
**Pull request by Lessorg**

### CODE REVIEW

File: src/main/java/test/project/bookshop/repository/book/BookRepository.java [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 import test.project.bookshop.model.Book;
 
 public interface BookRepository extends JpaRepository<Book, Long>,
         JpaSpecificationExecutor<Book> {
-
+    Page<Book> findAllByCategories_Id(Long categoryId, Pageable pageable);
```
Comment (sarakhmen) [inline src/main/java/test/project/bookshop/repository/book/BookRepository.java]: I'd recommend avoid using '_' within method names

**Review by sarakhmen — CHANGES_REQUESTED** [review]
great job! Just a small improvement required

File: src/main/java/test/project/bookshop/controller/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    @GetMapping("/{id}")
+    public CategoryDto getCategoryById(@PathVariable Long id) {
+        return categoryService.getById(id);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @Operation(
+            summary = "Create a new category",
+            description = "Create a new category")
+    @PostMapping
```
Comment (liliia-ponomarenko) [inline src/main/java/test/project/bookshop/controller/CategoryController.java:70]: ResponseStatus.CREATED

File: src/main/java/test/project/bookshop/model/Category.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Entity
+@Table(name = "categories")
+public class Category {
```
Comment (liliia-ponomarenko) [inline src/main/java/test/project/bookshop/model/Category.java:16]: Should it be a soft-deleted entity?

File: src/main/java/test/project/bookshop/dto/book/BookRequestDto.java [hunk 4]
```diff
@@ -20,4 +21,5 @@ public class BookRequestDto {
     private String coverImage;
     @Min(0)
     private BigDecimal price;
+    private List<Long> categoryIds;
```
Comment (liliia-ponomarenko) [inline src/main/java/test/project/bookshop/dto/book/BookRequestDto.java:24]: Can a book exist without Category? Let’s  use NotEmpty validation

File: src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -61,6 +61,15 @@ public BookDto update(Long id, BookRequestDto requestDto) {
         return bookMapper.toDto(bookRepository.save(book));
     }
 
+    @Override
+    public Page<BookDto> findBooksByCategotyId(Long id, Pageable pageable) {
+        Page<Book> bookPage = bookRepository.findAllByCategoriesId(id, pageable);
```
Comment (liliia-ponomarenko) [inline src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java:66]: you can use the method `map` of Page and simplify this method

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! A few recommendations ;)

**Pull request merged**